drift.py

```diff
@@ -1,146 +1,90 @@
 
 # Define SUCCESS for comparing command return codes.
 SUCCESS = 0
-args = None
-
+# Define result dataclass
+@dataclass
 class Result:
-    abstract:str
+    name:str
+    version:str
     commit:str
     tags:list
     timestamp:str
-    spec:str
-    version:str
-
-    def __init__(self):
-        self.abstract = ""
-        self.commit = ""
-        self.tags = []
-        self.timestamp = ""
-        self.spec = ""
-        self.version = ""
-
 
 class DriftAnalysis(Helicase):
-    specs = []
-    last = {}
-    def __init__(self, specs):
-        self.specs = specs
-        for spec in specs:
-            self.last[spec] = ""
+    def __init__(self, specs=None):
+        self.last = {}
+        self.specs = specs or []
 
     def analyze(self, commit):
-        for spec in self.specs:
-            out = spack("spec --json " + spec)
-            verToken = spec.find("@")
-            if verToken < 0: verToken = len(spec)
-            name = spec[:verToken]
-            result = Result()
-            result.commit = commit.hash
-            result.abstract = name
-            result.timestamp = str(commit.author_date)
-            if len(spec[verToken:]) > 0:
-                result.version = spec[verToken+1:]
-            if out.returncode == SUCCESS:
-                concrete_spec = json.loads(out.stdout)["spec"]
-                # In the case that the concretized spec hash changes
-                # we want to record the commit where this happens.
-                if self.last[spec] != "" and self.last[spec][0][name]["full_hash"] != concrete_spec[0][name]["full_hash"]:
-                    # Check if the package version is different.
-                    if self.last[spec][0][name]["version"] != concrete_spec[0][name]["version"]:
-                        result.tags += ["pkg-updated"]
-                    # +---------------------+
-                    # |     Dependencies    |
-                    # +---------------------+
-                    result.tags += set(self.tag_deps(concrete_spec, spec, name))
-                    # +---------------------+
-                    # |      Variants       |
-                    # +---------------------+
-                    result.tags += set(self.tag_variants(concrete_spec, spec, name))
-                    # Send Inflection Point
-                    send(result)
-                # Save concrete spec hash as last hash.
-                self.last[spec] = concrete_spec
-            else:
-                # If the spec doesn't concretize properly we also want
-                # to record the commit at which this occurred.
-                result.tags += ["concretizaton-failed"]
+        for abstract_spec in self.specs:
+            # Separate out spec version and name.
+            name = abstract_spec
+            version = ""
+            if "@" in abstract_spec:
+                name, version = abstract_spec.split("@", 1)
+            # Check that version exists in package.py file.
```

Comment:
I'm picky but it's ok to add empty lines before comments like this for readability.

Reply:
Another haha, @tgamblin you are always telling me to remove empty lines :P

Reply:
y'all are like complete opposites on this

Reply:
lol I think this might be true, I put a ton of spaces because it has to be visually pleasing to me!

Reply:
```python
"""
This is a docstring -- it has to be next to triple quotes but hates them so it scoots down a line

this is its description 

"""

Hello I am Vanessa

# below (but not just below) is some code

it is very beautiful and should be admired as if in a museum

# the line above is empty to facilitate admiration
```


```python
# I'm Alec what are docstrings let's get down to business
This is some code
# this describes the code below
this is some more code
my code likes other code so let's get close
# yes indeed
all done -- high signal. no noise or whitespace.
```

Reply:
lolol! I feel attacked.

Reply:
we can't attack you we're too busy admiring the code.

Reply:
And I'm keeping this screen shot forever. You truly understand me @tgamblin ! :laughing:

Reply:
Hahaha sooo I'll admit Vanessa told me you don't like spaces and thus I tried to limit them. I can go back to my more normal coding style and then you can push me to either more or less spaces/documentation.

Reply:
An example of non-compressed Alec coding lol
https://github.com/autamus/binoc/blob/main/parsers/spack.go

Reply:
oh I like spaces -- and Vanessa's are generally great b/c they're readable.  Sometimes too many for me but more is actually much better than less.